src/runtime.rs

```diff
@@ -78,12 +86,26 @@ impl Runtime {
                 }
             };
 
+            let wakeup = async {
```

Comment:
Why does this need to be in a timer and in park isn't sufficient? Is it because the `on_thread_park` callback isn't always invoked?

Reply:
from discord: I added that to make sure that we don't get starved in the case where we have futures getting woken by things which aren't io_uring. We could be starved if there was always work to do and if we don't push enough entries to force a flush of the SQ.

Reply:
100ms is pretty high... Could we add alternative hooks? Maybe when a task is done being polled?

Reply:
Splitting this into another PR